frontend/viewer/src/lib/components/audio/AudioDialog.svelte

```diff
@@ -80,11 +100,38 @@
   }
 
   async function onRecordingComplete(blob: Blob) {
-    selectedFile = undefined;
+    let fileExt = mimeTypeToFileExtension(blob.type);
+    selectedFile = new File([blob], `recording-${Date.now()}.${fileExt}`, {type: blob.type});
     if (!open) return;
     audio = await processAudio(blob);
   }
 
+  function mimeTypeToFileExtension(mimeType: string) {
+    if (mimeType.startsWith('audio/')) {
+      const baseType = mimeType.split(';')[0];
+      switch (baseType) {
+        case 'audio/mpeg':
+        case 'audio/mp3':
+          return 'mp3';
+        case 'audio/wav':
+        case 'audio/wave':
+        case 'audio/x-wav':
+          return 'wav';
+        case 'audio/ogg':
+          return 'ogg';
+        case 'audio/webm':
+          return 'webm';
+        case 'audio/aac':
+          return 'aac';
+        case 'audio/m4a':
```

Comment:
I've also come across `audio/mp4`... as well as `application/mp4` which we might need to watch out for.

Reply:
this is really only temporary as we'll be converting all files to a well know type once we get FFMPEG in place.